Makefile.am

```diff
@@ -10,6 +10,9 @@ EXTRA_PROGRAMS = CppUTestExtTests
 lib_LIBRARIES = lib/libCppUTest.a
 check_PROGRAMS = $(CPPUTEST_TESTS)
 
+# Change autoconf's hard-coded (and collision susceptible) HAVE_CONFIG_H macro.
+DEFS = $(filter-out -DHAVE_CONFIG_H,@DEFS@) -DCPPUTEST_HAVE_GENERATED_CONFIG_H
```

Comment:
Whoa....

Reply:
After a bit more checking on this, perhaps we can try out AX_PREFIX_CONFIG_H in the configure.ac.

My AI friend recommended to do this:
AC_CONFIG_HEADERS([config.h:config.hin], [AM_CPPFLAGS="$AM_CPPFLAGS -DCPPUTEST_HAVE_CONFIG_H"])

Which also feels like a hack... but at least a hack in the right place as this really shouldn't be in the am file :)

Reply:
> After a bit more checking on this, perhaps we can try out AX_PREFIX_CONFIG_H in the configure.ac.

I tried this and it only appears to affect the name of the resulting header, not the macro. We already have `m4/m4_ax_prefix_config_h.m4` to handle renaming the header.

> My AI friend recommended to do this:
AC_CONFIG_HEADERS([config.h:config.hin], [AM_CPPFLAGS="$AM_CPPFLAGS -DCPPUTEST_HAVE_CONFIG_H"])

I'll need to double check, but I think this only adds the `CPPUTEST_HAVE_CONFIG_H` macro; it doesn't remove the `HAVE_CONFIG_H` macro which we don't want leaking into other libraries.

Reply:
> My AI friend recommended to do this:
AC_CONFIG_HEADERS([config.h:config.hin], [AM_CPPFLAGS="$AM_CPPFLAGS -DCPPUTEST_HAVE_CONFIG_H"])

That doesn't build. I tried `AC_CONFIG_HEADERS([config.h], [AM_CPPFLAGS="$AM_CPPFLAGS -DCPPUTEST_HAVE_CONFIG_H"])` and confirmed that still defines `HAVE_CONFIG_H` with `DEFS = -DHAVE_CONFIG_H` in the generated makefile.